File: JAVA/2997. Minimum Number of Operations to Make Array XOR Equal to K.java
class Solution {
    public int minOperations(int[] nums, int k) {
        int xor = 0; 
        for (int num: nums) xor = (num ^ xor);
        xor = k ^ xor; 
        int ans = 0;
        while (xor != 0) {
            if ((xor&1)==1) ans++;
            xor = xor >> 1;
        }
        return ans;
    }
}

File: C++/2997. Minimum Number of Operations to Make Array XOR Equal to K.cpp
class Solution {
public:
    int minOperations(vector<int>& nums, int k) {
        int count = 0, x = 0;
        for (int i = 0; i < nums.size(); i++)
            x = x ^ nums[i];
        while (x > 0 || k > 0) {
            int a = x & 1, b = k & 1;
            if (a != b) count++; 
            x = x >> 1;
            k = k >> 1;
        }
        return count;
    }
};
